deweb-index/assembly/contracts/deweb-index.ts

```diff
@@ -0,0 +1,153 @@
+
+  setBytecode(args);
+}
+
+/**
+ * Change the owner of the smart contract to the address passed as an argument
+ * @param _binaryArgs - serialized arguments containing the new owner address
+ * @returns void
+ */
+export function transferOwnership(_binaryArgs: StaticArray<u8>): void {
+  _onlyOwner();
+
+  const args = new Args(_binaryArgs);
+  const newOwner = args
+    .nextSerializable<Address>()
+    .expect('Address is missing or invalid');
+
+  _setOwner(newOwner.toString());
+}
+
+/**
+ * Update the website address
+ * @param _binaryArgs - serialized arguments containing the new address
+ * @returns void
+ */
+export function updateWebsite(_binaryArgs: StaticArray<u8>): void {
+  const args = new Args(_binaryArgs);
+  const address = args
+    .nextSerializable<Address>()
+    .expect('Address is missing or invalid');
+
+  const version = Storage.getOf(address, DEWEB_VERSION_TAG);
+  assert(version.length > 0, 'The address is not a valid DEWEB address');
+
+  const owner = Storage.getOf(address, OWNER_KEY);
+  assert(owner.length > 0, 'Could not find the owner of the address');
+
+  _removeWebsite(address);
+  _addWebsite(address.toString(), owner);
+}
+
+/**
+ * Remove the website address
+ * @param _binaryArgs - serialized arguments containing the address
+ * @returns void
+ */
+export function removeWebsite(_binaryArgs: StaticArray<u8>): void {
+  _onlyOwner();
+
+  const args = new Args(_binaryArgs);
+  const address = args
+    .nextSerializable<Address>()
+    .expect('Address is missing or invalid');
+
+  _removeWebsite(address);
+}
+
+/**
+ * Entirely removes the website address from the storage.
+ * @param address - The website address to remove.
+ */
+function _removeWebsite(address: Address): void {
+  const keys = Storage.getKeys(stringToBytes(address.toString()));
+  for (let i = 0; i < keys.length; i++) {
+    const owner = bytesToString(keys[i]).slice(address.toString().length);
+
+    _removeWebsiteFromOwnerEnumeration(owner, address.toString());
+    Storage.del(keys[i]);
+  }
+}
+
+/**
+ * Adds the website address to the list of websites.
+ * @param address - The website address.
+ * @param owner - The owner's address.
+ */
+function _addWebsite(address: string, owner: string): void {
+  const key = stringToBytes(address).concat(stringToBytes(owner));
+
+  Storage.set(key, []);
+
+  _addWebsiteToOwnerEnumeration(owner, address);
+}
+
+/* -------------------------------------------------------------------------- */
+/*                                OWNER WEBSITE                               */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * Adds a website address to the owner's list of websites.
+ * @param owner - The owner's address.
+ * @param websiteAddress - The website address to add.
+ */
+function _addWebsiteToOwnerEnumeration(
+  owner: string,
+  websiteAddress: string,
+): void {
+  const key = stringToBytes(owner).concat(stringToBytes(websiteAddress));
```

Comment:
* add a prefix tag saying that this is the owner index so that in the future we can have multiple indexes without having to migrate the existing ones
* an attacker website can return an arbitrary string as "owner" so this concatenation is unsafe, and listing prefixes in those conditions is also super unsafe

Here is the format I would recommend:
`[INDEX_BY_OWNER_TAG][len(owner)][owner][website_address]`